smithy-model/src/main/java/software/amazon/smithy/model/traits/XmlNamespaceTrait.java

```diff
@@ -59,14 +70,19 @@ public static Builder builder() {
 
     @Override
     public Builder toBuilder() {
-        return builder().sourceLocation(getSourceLocation()).uri(uri);
+        Builder traitBuilder = builder()
+                .sourceLocation(getSourceLocation())
+                .uri(getUri());
+        getPrefix().ifPresent(traitBuilder::prefix);
```

Comment:
Given that you should allow nulls for this value in the builder anyways (it's optional), you could just chain here and pass in the potentially null property of `prefix`.